src/js/media-container.js

```diff
@@ -196,33 +196,46 @@ class MediaContainer extends window.HTMLElement {
   }
 
   mediaSetCallback(media) {
-    // Should only ever be set with a compatible media element, never null
-    if (!media) {
-      console.error('<media-chrome>: Media element set with slot="media" does not appear to be compatible.', media);
-      return false;
-    }
+    const resolveMediaPromise = (media) => {
```

Comment:
> mediaSetCallback was originally intended to mimic connectedCallback. As developer you're never responding to the result of connectedCallback, you're populating a known function property to be called when the element is connected.

Hmmm. Let me see if I understand your proposal:

- `media-container` handles the promise internally with a different function (like `handleMediaUpdated`)
- After media-container resolves the promise internally, then it calls `mediaSetCallback. If internal promise rejects then `mediaSetCallback` would never get called (instead of this proposed change which would reject the promise)
- so media-controller essentially doesn't need any refactoring because `mediaSetCallback` would no longer be returning a promise.

If I'm understanding this, it's all internal, I'm okay going either way with it.

Reply:
Yeah, exactly. We would just be removing the success check and `return` in media-controller's `if (!super.mediaSetCallback(media)) return;`, and anything that returns true or false in media-container's mediaSetCallback.